sonic_platform_base/sonic_xcvr/api/public/cmis.py

```diff
@@ -2092,11 +2098,271 @@ def set_application(self, channel, appl_code):
+            val = si_settings[consts.OUTPUT_EQ_PRE_CURSOR_TARGET_RX][si_key_lane]
+            if val <= self.get_rx_output_eq_pre_max_val():
+                if not self.nibble_read_modify_write(val, consts.OUTPUT_EQ_PRE_CURSOR_TARGET_RX, lane):
+                    return False
+        return True
+
+    def stage_output_eq_post_cursor_target_rx(self, host_lanes_mask, si_settings):
+        '''
+        This function applies RX output eq pre cursor settings
+        '''
+        for lane in range(self.NUM_CHANNELS):
+            if ((1 << lane) & host_lanes_mask) == 0:
+                continue
+            lane = lane+1
+            si_key_lane = "{}{}".format(consts.OUTPUT_EQ_POST_CURSOR_TARGET_RX, lane)
+            val = si_settings[consts.OUTPUT_EQ_POST_CURSOR_TARGET_RX][si_key_lane]
+            if val <= self.get_rx_output_eq_post_max_val():
+                if not self.nibble_read_modify_write(val, consts.OUTPUT_EQ_POST_CURSOR_TARGET_RX, lane):
+                    return False
+        return True            
+
+    def stage_output_amp_target_rx(self, host_lanes_mask, si_settings):
+        '''
+        This function applies RX output amp settings
+        '''
+        for lane in range(self.NUM_CHANNELS):
+            if ((1 << lane) & host_lanes_mask) == 0:
+                continue
+            lane = lane+1
+            si_key_lane = "{}{}".format(consts.OUTPUT_AMPLITUDE_TARGET_RX, lane)
+            val = si_settings[consts.OUTPUT_AMPLITUDE_TARGET_RX][si_key_lane]
+            if val <= self.get_rx_output_amp_supported_val():
+                if not self.nibble_read_modify_write(val, consts.OUTPUT_AMPLITUDE_TARGET_RX, lane):
+                    return False
+        return True
+
+    def stage_fixed_input_target_tx(self, host_lanes_mask, si_settings):
+        '''
+        This function applies fixed TX input si settings
+        '''
+        for lane in range(self.NUM_CHANNELS):
+            if ((1 << lane) & host_lanes_mask) == 0:
+                continue
+            lane = lane+1
+            si_key_lane = "{}{}".format(consts.FIXED_INPUT_EQ_TARGET_TX, lane)
+            val = si_settings[consts.FIXED_INPUT_EQ_TARGET_TX][si_key_lane]
+            if val <= self.get_tx_input_eq_max_val():
+                if not self.nibble_read_modify_write(val, consts.FIXED_INPUT_EQ_TARGET_TX, lane):
+                    return False
+        return True
+
+    def stage_adaptive_input_recall_tx(self, host_lanes_mask, si_settings):
+        '''
+        This function applies adaptive TX input recall si settings
+        '''
+        val = 0
+        for lane in range(self.NUM_CHANNELS):
+            if ((1 << lane) & host_lanes_mask) == 0:
+                continue
+            si_key_lane = "{}{}".format(consts.ADAPTIVE_INPUT_EQ_RECALLED_TX, lane+1)
+            si_val = si_settings[consts.ADAPTIVE_INPUT_EQ_RECALLED_TX][si_key_lane]
+            lane %= (self.NUM_CHANNELS//2)
```

Comment:
Please add comment here to describe all the calculations.

Reply:
taken care